src/lib/prov/tpm2/tpm2_rng.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+* TPM 2 RNG interface
+* (C) 2024 Jack Lloyd
+* (C) 2024 René Meusel, Amos Treiber - Rohde & Schwarz Cybersecurity GmbH
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/tpm2_rng.h>
+
+#include <botan/internal/stl_util.h>
+#include <botan/internal/tpm2_util.h>
+
+#include <source_location>
+#include <tss2/tss2_esys.h>
+
+namespace Botan {
+
+void TPM2_RNG::fill_bytes_with_input(std::span<uint8_t> output, std::span<const uint8_t> input) {
+   constexpr size_t MAX_STIR_RANDOM_SIZE = 128;  // From specification of tpm2-tool's tpm2_stirrandom
+
+   BufferSlicer in(input);
+
+   while(!in.empty()) {
+      TPM2B_SENSITIVE_DATA data;
+      data.size = std::min(in.remaining(), MAX_STIR_RANDOM_SIZE);
+      in.copy_into({data.buffer, data.size});
+
+      check_tss2_rc(
+         "StirRandom",
+         Esys_StirRandom(static_cast<ESYS_CONTEXT*>(m_ctx->get()), ESYS_TR_NONE, ESYS_TR_NONE, ESYS_TR_NONE, &data));
```

Comment:
Might be worth adding a helper in the internal util function that both performs the cast and also checks pointer validity.

```
inline ESYS_CONTEXT* inner(std::shared_ptr<TPM2_Context> ctx) {
   BOTAN_ASSERT_NOMSG(ctx != nullptr);
   void* inner = ctx->get();
   BOTAN_ASSERT_NOMSG(inner != nullptr);
   return static_cast<ESYS_CONTEXT*>(inner);
}
```

then

`Esys_StirRandom(inner(m_ctx), ESYS_TR_NONE, ...)`